ipam/http.go

```diff
@@ -33,16 +33,41 @@ func (alloc *Allocator) HandleHTTP(router *mux.Router) {
 		w.WriteHeader(204)
 	})
 
+	router.Methods("POST").Path("/ip/{id}/{ip}/{prefixlen}").HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		closedChan := w.(http.CloseNotifier).CloseNotify()
+		vars := mux.Vars(r)
+		ident := vars["id"]
+		cidr := vars["ip"] + "/" + vars["prefixlen"]
+		if addr, cidr, err := address.ParseCIDR(cidr); err != nil {
+			badRequest(w, err)
+			return
+		} else if addr == cidr.Start {
+			newAddr, err := alloc.AllocateInSubnet(ident, cidr, closedChan)
+			if err != nil {
+				badRequest(w, fmt.Errorf("Unable to allocate: %s", err))
+				return
+			}
+
+			fmt.Fprintf(w, "%s/%d", newAddr.String(), cidr.PrefixLen)
+		} else {
+			fmt.Fprintf(w, "%s/%d", addr.String(), cidr.PrefixLen)
+		}
+	})
+
```

Comment:
I'd turn this into
````
addr, cidr, err := ...
if err != nil {
    ....
}
if addr == cidr.Start {
...
}
fmt.Fprintf(w, "%s/%d", addr.String(), cidr.PrefixLen)
````

Also, perhaps turn the result into a `CIDR`, which ought to produce stringify to the intended result.